apps/vmq_diversity/src/vmq_diversity_bcrypt.erl

```diff
@@ -30,6 +30,7 @@ gen_salt(_, St) ->
     {ok, Salt} = bcrypt:gen_salt(),
     {[list_to_binary(Salt)], St}.
 
+-dialyzer({nowarn_function, hashpw/2}).
```

Comment:
Curious what is dialyzer complaining about here?

Reply:
The call bcrypt:hashpw(Pass::binary(), Salt::binary()) will never return since it differs in the 2nd argument from the success typing arguments: (binary() | [byte()], [byte()])

https://github.com/erlangpack/bcrypt/blob/master/src/bcrypt.erl
-spec hashpw( Password, Salt ) -> Result when
	Password :: [byte()] | binary(), 
	Salt :: [byte()] | binary(),
	Result :: {ok, Hash} | {error, ErrorDescription},
	Hash :: [byte()],
	ErrorDescription :: pwerr().